src/Sentry/HubExtensions.cs

```diff
@@ -1,96 +1,90 @@
-        /// <param name="level">Breadcrumb level</param>
+        /// <param name="hub">The Hub which holds the scope stack.</param>
+        /// <param name="message">The message.</param>
+        /// <param name="category">Category.</param>
+        /// <param name="type">Breadcrumb type.</param>
+        /// <param name="data">Additional data.</param>
+        /// <param name="level">Breadcrumb level.</param>
         public static void AddBreadcrumb(
             this IHub hub,
-            string message,
-            string category = null,
-            string type = null,
-            IDictionary<string, string> data = null,
+            string? message,
+            string? category = null,
+            string? type = null,
+            IDictionary<string, string>? data = null,
             BreadcrumbLevel level = default)
             => hub.AddBreadcrumb(
-                clock: null,
-                message: message,
-                type: type,
-                data: data != null ? new Dictionary<string,string>(data) : null,
-                category: category,
-                level: level);
+                null,
+                message,
+                category,
+                type,
+                data != null ? new Dictionary<string, string>(data) : null,
+                level);
 
         /// <summary>
-        /// Adds a breadcrumb using a custom <see cref="ISystemClock"/> which allows better testability
+        /// Adds a breadcrumb using a custom <see cref="ISystemClock"/> which allows better testability.
         /// </summary>
-        /// <param name="hub">The Hub which holds the scope stack</param>
-        /// <param name="clock">The system clock</param>
-        /// <param name="message">The message</param>
-        /// <param name="category">Category</param>
-        /// <param name="type">Breadcrumb type</param>
-        /// <param name="data">Additional data</param>
-        /// <param name="level">Breadcrumb level</param>
+        /// <param name="hub">The Hub which holds the scope stack.</param>
+        /// <param name="clock">The system clock.</param>
+        /// <param name="message">The message.</param>
+        /// <param name="category">Category.</param>
+        /// <param name="type">Breadcrumb type.</param>
+        /// <param name="data">Additional data.</param>
+        /// <param name="level">Breadcrumb level.</param>
         /// <remarks>
-        /// This method is to be used by integrations to allow testing
+        /// This method is to be used by integrations to allow testing.
         /// </remarks>
         [EditorBrowsable(EditorBrowsableState.Never)]
         public static void AddBreadcrumb(
             this IHub hub,
-            ISystemClock clock,
-            string message,
-            string category = null,
-            string type = null,
-            IDictionary<string, string> data = null,
+            ISystemClock? clock,
+            string? message,
```

Comment:
There are tests that set it to null. Should those be removed? I would prefer to remove the tests in a separate PR because then there's a risk I might break something.

Reply:
I'd say either we get the tests to run/compile with net461/netcore2 and pass null anyway (as discussed above we can't guarantee consumers will enable C# 8 and have polyfills), that'd be best.